pkg/kubelet/kubelet.go

```diff
@@ -1575,6 +1585,13 @@ func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 		if err := handler.SyncPods(pods, podSyncTypes, start); err != nil {
 			glog.Errorf("Couldn't sync containers: %v", err)
 		}
+
+		syncedCount++
+		if kl.statusUpdateFrequency != 0 && kl.kubeClient != nil && syncedCount%kl.statusUpdateFrequency == 0 {
```

Comment:
One major reason is to support disable sync (set updatefrequency to 0).  The other is that the method becomes a bit complex.  Considering we are posting pod status too, I tend to move this out of main syncloop